src/js/app/service/delivery.js

```diff
@@ -0,0 +1,165 @@
+ */
+define([
+    'lodash',
+    'core/store',
+    'lib/uuid'
+], function(_, store, uuid){
+    'use strict';
+
+    var deliveryStoreName = 'delivery';
+
+    /**
+     * Definition of a minimal delivery as seen by the service
+     * @typedef {Object} delivery
+     * @param {String} id - the delivery unique identifier
+     * @param {String} label - the delivery label
+     */
+
+    /**
+     * Validates a delivery object
+     * @param {delivery} delivery
+     * @returns {Boolean} true if valid
+     * @throws {TypeError} when the delivery doesn't match the expectations
+     */
+    var validateDelivery = function validateDelivery(delivery){
+        if(!_.isPlainObject(delivery)){
+            throw new TypeError('Missing or invalid delivery');
+        }
+        if(_.isEmpty(delivery.id)){
+            throw new TypeError('An delivery needs to have a property id');
+        }
+        if(_.isEmpty(delivery.label)){
+            throw new TypeError('A delivery needs to have a property label');
+        }
+        return true;
+    };
+
+    /**
+     * @typedef {Object} deliveryService
+     */
+    return {
+
+        /**
+         * Get a delivery from it's identifier / URI.
+         *
+         * @param {String} id - the delivery identifier
+         * @returns {Promise<delivery>} resolves with the delivery or null if not found
+         */
+        getById : function getById(id){
+            if(_.isEmpty(id)){
+                return Promise.resolve(null);
+            }
+            return store(deliveryStoreName).then(function(deliveryStore){
+                return deliveryStore.getItem(id);
+            });
+        },
+
+        /**
+         * Get all deliveries
+         * @returns {Promise<Object[]>} resolves with the delivery collection
+         */
+        getAll : function getAll(){
+            return store(deliveryStoreName).then(function(deliveryStore){
+                return deliveryStore.getItems();
+            });
+        },
+
+        /**
+         * Set an delivery, if one exists already, it will be replaced !
+         * @param {delivery} delivery
+         * @returns {Promise<Boolean>} resolves with true if set
+         */
+        set : function set(delivery){
+
+            validateDelivery(delivery);
+
+            return store(deliveryStoreName).then(function(deliveryStore){
+                return deliveryStore.setItem(delivery.id, delivery);
+            });
+        },
+
+        /**
+         * Update an delivery,
+         * if an entry was already there, we will merge them,
+         * using the existing values as default.
+         *
+         * @param {delivery} delivery - the delivery to update
+         * @returns {Promise<Boolean>} resolves with true if updated
+         */
+        update : function update(delivery){
+            var self = this;
+
+            if(!_.isPlainObject(delivery) || !delivery.id){
+                return Promise.resolve(false);
```

Comment:
Yes.
May be the method names doesn't reflect enough the intents, but 

-  `set` is intended to assign (add or replace) a valid object, here a valid delivery. If you set something that is not a delivery we just throw (this can't be a user error), a type error is thrown because we want to warn the developer he missyuse the API.

- `update` is intended to merge (or update) an existing object. You don't need to give in parameter the full delivery object, you can update only some properties 

For example

```
deliveryService.set({ id : '123', label : 'foo' }); 
//here if the label is not set, the method throws because a delivery must have a label.

deliveryService.update({ id : '123', updateDate : new Date() });
//here we merge the values from anything in parameter.
```
Maybe the name `update` is missleading